### PR TITLE
Generalize Interpolation routines in diagnostics for anisotropic ref ratio

### PR DESCRIPTION
**Pull request by prkkumar**
This PR generalizes the interpolation routines in diagnostics for anisotropic refinement ratio.

### CODE REVIEW

File: Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp [hunk 1]
```diff
@@ -467,7 +467,8 @@ WriteCoarseVector( const std::string field_name,
         WriteZeroRawMF( *Fz_fp, dm, filename, level_prefix, field_name+"z_cp", lev, ng );
     } else {
         // Interpolate coarse data onto fine grid
-        const int r_ratio = WarpX::GetInstance().refRatio(lev-1)[0];
+        amrex::IntVect r_ratio = WarpX::GetInstance().refRatio(lev-1);
+        //const int r_ratio = WarpX::GetInstance().refRatio(lev-1)[0];
```
Comment (EZoni) [inline Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp]:
```suggestion
```

File: Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp [hunk 2]
```diff
@@ -504,7 +505,8 @@ WriteCoarseScalar( const std::string field_name,
         // Create an alias to the component `icomp` of F_cp
         MultiFab F_comp(*F_cp, amrex::make_alias, icomp, 1);
         // Interpolate coarse data onto fine grid
-        const int r_ratio = WarpX::GetInstance().refRatio(lev-1)[0];
+        //const int r_ratio = WarpX::GetInstance().refRatio(lev-1)[0];
```
Comment (EZoni) [inline Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp]:
```suggestion
```

File: Source/Utils/Interpolate.H [hunk 3]
```diff
@@ -18,7 +18,7 @@ namespace Interpolate
     std::unique_ptr<MultiFab>
         getInterpolatedScalar(
             const MultiFab& F_cp, const MultiFab& F_fp,
-            const DistributionMapping& dm, const int r_ratio,
+            const DistributionMapping& dm, const IntVect r_ratio,
```
Comment (EZoni) [inline Source/Utils/Interpolate.H]:
```suggestion
            const DistributionMapping& dm, const amrex::IntVect r_ratio,
```

File: Source/Utils/Interpolate.H [hunk 4]
```diff
@@ -32,7 +32,7 @@ namespace Interpolate
             const MultiFab* Fx_fp,
             const MultiFab* Fy_fp,
             const MultiFab* Fz_fp,
-            const DistributionMapping& dm, const int r_ratio,
+            const DistributionMapping& dm, const IntVect r_ratio,
```
Comment (EZoni) [inline Source/Utils/Interpolate.H]:
```suggestion
            const DistributionMapping& dm, const amrex::IntVect r_ratio,
```

File: Source/Utils/Interpolate.cpp [hunk 5]
```diff
@@ -25,7 +25,7 @@ namespace Interpolate
     std::unique_ptr<MultiFab>
     getInterpolatedScalar(
         const MultiFab& F_cp, const MultiFab& F_fp,
-        const DistributionMapping& dm, const int r_ratio,
+        const DistributionMapping& dm, const IntVect r_ratio,
```
Comment (EZoni) [inline Source/Utils/Interpolate.cpp]:
```suggestion
        const DistributionMapping& dm, const amrex::IntVect r_ratio,
```

File: Source/Utils/Interpolate.cpp [hunk 6]
```diff
@@ -49,7 +49,7 @@ namespace Interpolate
 
                 // - Fully nodal
                 if ( F_fp.is_nodal() ){
-                    IntVect refinement_vector{AMREX_D_DECL(r_ratio, r_ratio, r_ratio)};
+                    IntVect refinement_vector{AMREX_D_DECL(r_ratio[0], r_ratio[1], r_ratio[2])};
```
Comment (EZoni) [inline Source/Utils/Interpolate.cpp]:
```suggestion
                    amrex::IntVect refinement_vector{AMREX_D_DECL(r_ratio[0], r_ratio[1], r_ratio[2])};
```

File: Source/Utils/Interpolate.cpp [hunk 7]
```diff
@@ -72,7 +72,7 @@ namespace Interpolate
         const MultiFab* Fx_fp,
         const MultiFab* Fy_fp,
         const MultiFab* Fz_fp,
-        const DistributionMapping& dm, const int r_ratio,
+        const DistributionMapping& dm, const IntVect r_ratio,
```
Comment (EZoni) [inline Source/Utils/Interpolate.cpp]:
```suggestion
        const DistributionMapping& dm, const amrex::IntVect r_ratio,
```

File: Source/Utils/Interpolate_K.H [hunk 8]
```diff
@@ -9,18 +9,19 @@ AMREX_GPU_DEVICE AMREX_FORCE_INLINE
 void interp (int j, int k, int l,
              amrex::Array4<amrex::Real      > const& fine,
              amrex::Array4<amrex::Real const> const& crse,
-             int r_ratio, IntVect const& type) noexcept
+             IntVect r_ratio, IntVect const& type) noexcept
```
Comment (EZoni) [inline Source/Utils/Interpolate_K.H]:
```suggestion
             const amrex::IntVect r_ratio, IntVect const& type) noexcept
```

**Review by EZoni — CHANGES_REQUESTED** [review]
Thank you for this PR, @prkkumar! It looks good to me, I left only a few suggestions.

One question:
It seems that no checksums need to be modified, does it mean that no CI test is currently using an anisotropic refinement ratio? Otherwise, I would expect some checksums  to change, if this interpolation routines are actually used for diagnostics.

**Comment by prkkumar** [comment]
> 

Thanks for reviewing, @EZoni! I have applied the suggestions. 

I am working on a separate PR that adds a CI test using an anisotropic refinement ratio.

**Review by EZoni — APPROVED** [review]

**Review by atmyers — APPROVED** [review]

**Pull request merged**